### DISCUSSION TITLE
C# Lambda with AWS IAM Auth method error in certificate

### DISCUSSION THREAD

**Discussion by selvaesakkirajan** [rajanadar/VaultSharp#359] [Q&A]
I am creating C# lambda 
// setup the AWS-IAM based auth to get the right token.
var amazonSecurityTokenServiceConfig = new AmazonSecurityTokenServiceConfig();
var iamRequest = GetCallerIdentityRequestMarshaller.Instance.Marshall(new GetCallerIdentityRequest());
iamRequest.Endpoint = new Uri(amazonSecurityTokenServiceConfig.DetermineServiceURL());
iamRequest.ResourcePath = "/";

iamRequest.Headers.Add("User-Agent", httpvaultaddress);
iamRequest.Headers.Add("X-Amz-Security-Token", AWS_SESSION_TOKEN);
iamRequest.Headers.Add("Content-Type", "application/x-www-form-urlencoded; charset=utf-8");

new AWS4Signer().Sign(iamRequest, amazonSecurityTokenServiceConfig, new RequestMetrics(), AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY);

var iamSTSRequestHeaders = iamRequest.Headers;
var base64EncodedIamRequestHeaders = Convert.ToBase64String(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(iamSTSRequestHeaders)));
IAuthMethodInfo authMethod = new IAMAWSAuthMethodInfo(nonce: System.Guid.NewGuid().ToString(), roleName: roleName, requestHeaders: base64EncodedIamRequestHeaders);
var vaultClientSettings = new VaultClientSettings(httpVaultAddress, authMethod);

IVaultClient vaultClient = new VaultClient(vaultClientSettings);

vaultClient.V1.Auth.PerformImmediateLogin();

Above login throws an error:
The SSL connection could not be established, see inner exception.
The remote certificate is invalid because of errors in the certificate chain:UntrustedRoot

Can you help in fixing this

**Comment by konidev20** [comment]
@selvaesakkirajan, 

* Are you facing this issue in your local development environment?
* Are you running `vault` server in HTTPS mode? 
* Are you able to perform the same operation using the `vault` CLI? 

About the error you're getting: 
* This is common when you're using dev certificates or certificates without a CA. 
* If you're running a local instance of `vault` for testing, see if you can create a local self-signed CA certificate using `openssl` and then sign the certificate of your `vault` server. Post that you can try the operation. [Create self-signed certificates using openssl](https://devopscube.com/create-self-signed-certificates-openssl/)

**Comment by rajanadar** [comment]
@selvaesakkirajan  This is outside of VaultSharp and @konidev20  has suggested the solution correctly.
